File: src/main/java/me/sul/crackshotaddition/events/WeaponSwapEvent.java
package me.sul.crackshotaddition.events;

import lombok.Getter;
import org.bukkit.entity.Player;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.bukkit.inventory.ItemStack;

@Getter
public class WeaponSwapEvent extends Event {
    private static final HandlerList handlers = new HandlerList();
    private final Player player;
    private final ItemStack weaponItem;
    private final String weaponTitle;
    private final int swapDelay;

    public WeaponSwapEvent(Player player, ItemStack weaponItem, String weaponTitle, Integer swapDelay) {
        this.player = player;
        this.weaponItem = weaponItem;
        this.weaponTitle = weaponTitle;
        this.swapDelay = swapDelay;
    }

    @Override
    public HandlerList getHandlers() {
        return handlers;
    }
    public static HandlerList getHandlerList()
    {
        return handlers;
    }
}

File: src/main/kotlin/me/sul/crackshotaddition/events/WeaponSwapEvent.kt
package me.sul.crackshotaddition.events

import org.bukkit.entity.Player
import org.bukkit.event.Event
import org.bukkit.event.HandlerList
import org.bukkit.inventory.ItemStack

class WeaponSwapEvent(val player: Player, val weaponItem: ItemStack, val weaponTitle: String, val swapDelay: Int) : Event() {
    override fun getHandlers(): HandlerList {
        return handlerList
    }

    companion object {
        val handlerList = HandlerList()
    }
}